cluster-logging-operator/pkg/apis/logging/v1alpha1/types.go

```diff
@@ -0,0 +1,110 @@
+package v1alpha1
+
+import (
+  metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterLoggingList struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ListMeta   `json:"metadata"`
+  Items             []ClusterLogging `json:"items"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterLogging struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata"`
+  Spec              ClusterLoggingSpec   `json:"spec"`
+  Status            ClusterLoggingStatus `json:"status,omitempty"`
+}
+
+type ClusterLoggingSpec struct {
+  AllInOne          bool `json:allinone`
+  Visualization     VisualizationSpec `json:"visualization"`
+  LogStore          LogStoreSpec `json:"logStore"`
+  Collection        CollectionSpec `json:"collection"`
+  Curation          CurationSpec `json:"curation"`
+}
+
+// This is the struct that will contain information pertinent to Log visualization (Kibana)
+type VisualizationSpec struct {
+  Disabled          bool `json:"disabled"`
+  Resources         v1.ResourceRequirements `json:"resources"`
+  NodeSelector      map[string]string `json:"nodeSelector,omitempty"`
+  Replicas          int32 `json:"replicas"`
+}
+
+// This is the struct that will contain information pertinent to Log storage (Elasticsearch)
+type LogStoreSpec struct {
+  Disabled          bool `json:"disabled"`
+  Resources         v1.ResourceRequirements `json:"resources"`
+  NodeSelector      map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
Should we have a storage option here at all? We expose that on the elasticsearch operator side currently https://github.com/ViaQ/elasticsearch-operator/blob/master/pkg/apis/elasticsearch/v1alpha1/types.go#L33